### PR TITLE
docs(README): updated and added the ConditionalPlugin 

### PR DESCRIPTION
**Pull request by raghav4**
issue: https://github.com/accordproject/cicero-ui/issues/145

# Issue #145 
* Added the [ConditionalPlugin](https://github.com/accordproject/cicero-ui/blob/master/src/plugins/ConditionalPlugin.js) 

### Changes

- Added ConditionalPlugin
- Fixed markdown warnings, by fixing spaces.

### CODE REVIEW

File: src/plugins/README.md [hunk 1]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+    onPaste,
+    queries: {
+      findClauseNode,
+      isOutsideOfClause: isEditable,
+      isClauseSupported
+    }
+}
+```
+
+where augmentSchema, isEditable and renderInline are functions.
```
Comment (DianaLease) [inline src/plugins/README.md]: `augmentSchema, renderBlock, isEditable, onChange, onPaste, findClauseNode, isOutsideOfClause, and isClauseSupported` are all functions

Comment (raghav4) [reply src/plugins/README.md]: adding it @DianaLease!

File: src/plugins/README.md [hunk 2]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+    onPaste,
+    queries: {
+      findClauseNode,
+      isOutsideOfClause: isEditable,
+      isClauseSupported
+    }
+}
+```
+
+where augmentSchema, renderBlock, isEditable, onChange, onPaste, findClauseNode, isOutsideOfClause, and isClauseSupported are functions.
```
Comment (irmerk) [inline src/plugins/README.md]: Change to "where `augmentSchema`, `renderBlock`, `isEditable`, `onChange`, `onPaste`, `findClauseNode`, `isOutsideOfClause`, and `isClauseSupported` are functions."

Also provide explanation of these functions.

File: src/plugins/README.md [hunk 3]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+
+## ConditionalPlugin
+
+### What it does?
```
Comment (irmerk) [inline src/plugins/README.md]: Could add this below `ClausePlugin` for consistency.

File: src/plugins/README.md [hunk 4]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+```js
+{
+  name,
+  augmentSchema,
+  isEditable,
+  renderInline,
+}
+```
+
+where augmentSchema, isEditable and renderInline are functions.
```
Comment (irmerk) [inline src/plugins/README.md]: Change to "where `augmentSchema`, `isEditable`, and `renderInline ` are functions."

File: src/plugins/README.md [hunk 5]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+
+### augmentSchema function
```
Comment (irmerk) [inline src/plugins/README.md]: Change to `augmentSchema` function

File: src/plugins/README.md [hunk 6]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+
+augmentSchema function returns a new schema for the addition of conditional type in the markdown editor in the format suggested by [slate.js](https://docs.slatejs.org/).
+
+### isEditable function
```
Comment (irmerk) [inline src/plugins/README.md]: Change to `isEditable` function

File: src/plugins/README.md [hunk 7]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+
+isEditable function returns a boolean on checking whether the conditional field in the markdown editor is editable or not.
+
+### renderInline function
```
Comment (irmerk) [inline src/plugins/README.md]: Change to `renderInline` function

File: src/plugins/README.md [hunk 8]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+
+where augmentSchema, isEditable and renderInline are functions.
+
+### augmentSchema function
+
+augmentSchema function returns a new schema for the addition of conditional type in the markdown editor in the format suggested by [slate.js](https://docs.slatejs.org/).
+
+### isEditable function
+
+isEditable function returns a boolean on checking whether the conditional field in the markdown editor is editable or not.
```
Comment (irmerk) [inline src/plugins/README.md]: Change to `isEditable`

File: src/plugins/README.md [hunk 9]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+
+### renderInline function
+
+renderInline function renders a conditional inline to the text in the editor by returning a span tag with the props.
```
Comment (irmerk) [inline src/plugins/README.md]: Change to `renderInline`

File: src/plugins/README.md [hunk 10]
```diff
@@ -174,3 +165,76 @@ const plugins = [ClausePlugin()];
...
+  isEditable,
+  renderInline,
+}
+```
+
+where augmentSchema, isEditable and renderInline are functions.
+
+### augmentSchema function
+
+augmentSchema function returns a new schema for the addition of conditional type in the markdown editor in the format suggested by [slate.js](https://docs.slatejs.org/).
```
Comment (irmerk) [inline src/plugins/README.md]: Change to `augmentSchema`

**Comment by raghav4** [comment]
@irmerk fixing it asap!

File: src/plugins/README.md (hunk 4 above)
Comment (raghav4) [reply src/plugins/README.md]: Done

**Comment by raghav4** [comment]
> @raghav4 Please update your branch and squash your commits, along with that do run these from your local branch :
> 
> ```
> git commit --amend --signoff
> ```
> 
> ```
> git push --force-with-lease origin <local-branch-name>
> ```

@elit-altum I tried but it didn't work, used the same commands

**Comment by elit-altum** [comment]
> > @raghav4 Please update your branch and squash your commits, along with that do run these from your local branch :
> > ```
> > git commit --amend --signoff
> > ```
> > 
> > 
> > ```
> > git push --force-with-lease origin <local-branch-name>
> > ```
> 
> @elit-altum I tried but it didn't work, used the same commands

It might be because you tried to ```force-with-lease``` a commit amend which you previously 'forced' . Try using the ```--force``` command instead of it to while commiting the amended message.

**Comment by raghav4** [comment]
@elit-altum didn't work :(

**Comment by raghav4** [comment]
Hello @irmerk, I've raised a new PR https://github.com/accordproject/cicero-ui/pull/305, because this one is having issues while committing as one of my commits isn't signed. I couldn't figure it which one was it and I've tried every way possible but it isn't working. So, I've raised a new PR. You could check it out and let me know if there's anything left. Though I've done whatever you've suggested I didn't get this https://github.com/accordproject/cicero-ui/pull/296#discussion_r383513142

**Comment by elit-altum** [comment]
> Hello @irmerk, I've raised a new PR #305, because this one is having issues while committing as one of my commits isn't signed. I couldn't figure it which one was it and I've tried every way possible but it isn't working. So, I've raised a new PR. You could check it out and let me know if there's anything left. Though I've done whatever you've suggested I didn't get this [#296 (comment)](https://github.com/accordproject/cicero-ui/pull/296#discussion_r383513142)

@raghav4 Please avoid this in future. If everything fails you should make a local copy of your changes and use ```git reset -- hard``` with respect to upstream master and then make a new commit by adding your local changes.
✌️

**Comment by raghav4** [comment]
@elit-altum I tried but there were too many merge conflicts so I avoided it.

**Comment by irmerk** [comment]
Closing for #305

**Pull request closed without merging**